src/test/cls_lock/test_cls_lock.cc

```diff
@@ -502,10 +502,11 @@ TEST(ClsLock, TestExclusiveEphemeralBasic) {
   ASSERT_EQ(0, l1.lock_exclusive_ephemeral(&ioctx, oid1));
   ASSERT_EQ(0, ioctx.stat(oid1, &size, &mod_time));
   sleep(2);
+  bool r1_expired = lock_expired(&ioctx, oid1, lock_name1);
   int r1 = l1.unlock(&ioctx, oid1);
-  EXPECT_TRUE(r1 == 0 || ((r1 == -ENOENT) && (lock_expired(&ioctx, oid1, lock_name1))))
+  EXPECT_TRUE(r1 == 0 || ((r1 == -ENOENT) && (r1_expired)))
     << "unlock should return 0 or -ENOENT return: " << r1;
-  ASSERT_EQ(-ENOENT, ioctx.stat(oid1, &size, &mod_time));
+  ASSERT_EQ((((r1 == -ENOENT) && (r1_expired)) ? 0 : -ENOENT), ioctx.stat(oid1, &size, &mod_time));
```

Comment:
That said, I'm having a hard time wrapping my head around this test to begin with.  Why is it setting the lock duration to 5 seconds but sleeping only for 2 seconds and then attempts to cover both "automatically expired" and "manually unlocked" cases in the same asserts?  These cases should be separated.

It looks like some of this was introduced in https://github.com/ceph/ceph/pull/45762.  I think that fixup was wrong, instead it should have probably just increased the lock duration from 5 seconds to e.g. 5 minutes to avoid timing irregularities.

Adding @badone who approved https://github.com/ceph/ceph/pull/45762 and @ivancich who implemented exclusive ephemeral locks.

Reply:
@idryomov The test will fail if we have some delay in the test, and yes, increase the lock duration will solve the failure if we have some delays during the test run via teuthology.
i'll change the lock duration instead of checking if the lock expired